### ISSUE TITLE
Review handling of failures in CommandHandlerVerticle.cmdHandler

### ISSUE THREAD

**Issue by juanavelez** [crabzilla/crabzilla#22]
I think there are places within this method that don't fail the future parameter. For example consider the scenario where eventRepository.selectAfterVersion fails midway through the execution (e.g. connection failure, MySQL went down, etc.). The handling right now is to throw an RTE which does nothing to fail the future parameter. I'd suggest a) changing the eventRepository methods to accept futures and b) using another future that on failure will fail the future parameter and on success will call the current handler

```java
      eventRepository.selectAfterVersion(targetId, cachedSnapshot.getVersion(), nonCached -> {

        val totalOfNonCachedEvents = nonCached.getEvents().size();
```
for
```java
     Future<SnapshotData> future2 = Future.future();
     eventRepository.selectAfterVersion(targetId, cachedSnapshot.getVersion(), future2);
     future2.setHandler( ar -> {
     if (ar.fail( )
       future1.fail(ar.cause());
     else {
       nonCached = ar.result();
       val totalOfNonCachedEvents = nonCached.getEvents().size();
```

**Comment by rodolfodpk** [comment]
Thanks a lot, @juanavelez

**Comment by juanavelez** [comment]
Fixed by #23
